File: state_adapter/src/main/java/me/simple/state_adapter/StateViewHolder.java
package me.simple.state_adapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;
import android.view.View;
import android.view.ViewGroup;

public class StateViewHolder extends RecyclerView.ViewHolder {

    public StateViewHolder(@NonNull View itemView) {
        super(itemView);
        ViewGroup.LayoutParams params = itemView.getLayoutParams();
        if (params instanceof StaggeredGridLayoutManager.LayoutParams) {
            ((StaggeredGridLayoutManager.LayoutParams) params).setFullSpan(true);
        }
    }

    public void setState(int state) {

    }
}

File: state_adapter/src/main/java/me/simple/state_adapter/StateViewHolder.kt
package me.simple.state_adapter

import android.view.View
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import androidx.recyclerview.widget.StaggeredGridLayoutManager

class StateViewHolder(itemView: View) : ViewHolder(itemView) {

    init {
        val params = itemView.layoutParams
        if (params is StaggeredGridLayoutManager.LayoutParams) {
            params.isFullSpan = true
        }
    }
}